src/engines/nwn2/journal.cpp

```diff
@@ -0,0 +1,86 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  A journal in a Neverwinter Nights 2 module.
+ */
+
+#include <algorithm>
+
+#include "src/aurora/gff3file.h"
+
+#include "src/engines/nwn2/journal.h"
+
+namespace Engines {
+
+namespace NWN2 {
+
+Journal::Journal(const Aurora::GFF3Struct &journal) {
+
+	load(journal);
+}
+
+Journal::~Journal() {
+}
+
+uint32 Journal::getJournalQuestExperience(const Common::UString plotID) const {
+	auto category = std::find_if(_categories.begin(), _categories.end(), [&](const Category &c) {
+		return c.name == plotID;
+	});
+
+	return (category != _categories.end()) ? category->XP : 0;
+}
+
+void Journal::load(const Aurora::GFF3Struct &journal) {
+	if (!journal.hasField("Categories"))
+		return;
+
+	const Aurora::GFF3List &categories = journal.getList("Categories");
+	for (Aurora::GFF3List::const_iterator c = categories.begin(); c != categories.end(); ++c) {
```

Comment:
You can use a range-based for-loop for iterating through a GFF3List as well.

Should be
```cpp
for (const Aurora::GFF3Struct *c : categories) {
}
```